tests/Feature/BookingTest.php

```diff
@@ -0,0 +1,131 @@
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Foundation\Testing\WithFaker;
+use Illuminate\Testing\Fluent\AssertableJson;
+use Tests\TestCase;
+
+class BookingTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    public function test_bookRoom_success(): void
+    {
+        Booking::factory()->create();
+        Room::factory()->create();
+
+        $testData = [
+            "room_id" => 1,
+            "customer" => "Test customer",
+            "guests" => 2,
+            "start" => "2024-12-15",
+            "end" => "2024-12-25"
+        ];
+
+        $response = $this->post('/api/bookings', $testData);
+
+        $response->assertStatus(201)
+        ->assertJson(function (AssertableJson $json){
+            $json->hasAll(['message', 'data'])
+            ->has('data', function (AssertableJson $data) {
+                $data->hasAll([
+                    'room_id',
+                    'customer',
+                    'start',
+                    'end',
+                    'updated_at',
+                    'created_at',
+                    'id'
+                ])->whereAllType([
+                    'room_id' => 'integer',
+                    'customer' => 'string',
+                    'start' => 'string',
+                    'end' => 'string',
+                    'updated_at' => 'string|null',
+                    'created_at' => 'string|null',
+                    'id' => 'integer'
+                ]);
+            });
+        });
+    }
+
+    public function test_bookRoom_failure_dueTo_requiredFieldsMissing()
+    {
+        $testData = [];
+
+        $response = $this->postJson('/api/bookings', $testData);
+
+        $response->assertStatus(422)
+            ->assertInvalid(['room_id', 'customer', 'guests', 'start', 'end'])
+            ->assertJson(function (AssertableJson $json){
+                $json->hasAll('message', 'errors');
+        });
+    }
+
+    public function test_bookRoom_failure_dueTo_incorrectRoomCapacity()
+    {
+        Room::factory()->create();
+
+        $testData = [
+            "room_id" => 1,
+            "customer" => "Test customer",
+            "guests" => 500,
+            "start" => "2024-12-15",
+            "end" => "2024-12-25"
+        ];
+
+        $response = $this->post('/api/bookings', $testData);
+
+        $response->assertStatus(400)
+            ->assertJson(function (AssertableJson $json){
+                $json->hasAll(['message']);
+        });
+    }
+
+    public function test_bookRoom_failure_dueTo_endDateBeforeStartDate()
+    {
+        Booking::factory()->create();
+        Room::factory()->create();
+
+        $testData = [
+            "room_id" => 1,
+            "customer" => "Test customer",
+            "guests" => 2,
+            "start" => "2026-12-25",
+            "end" => "2026-12-01"
+        ];
+
+        $response = $this->postJson('/api/bookings', $testData);
+
+        $response->assertStatus(422) // Is 422 ok here???? //
```

Comment:
Yeah 422 is ok given your current solution for checking the dates. The api docs does ask for a 400 with a custom message (one not generated by validation), but I don't mind if you want to stick with what you have now